### PR TITLE
DeviceSerialized.__reduce_ex__: convert frame to numpy arrays

### PR DESCRIPTION
**Pull request by madsbk**

### CODE REVIEW

File: dask_cuda/device_host_file.py [hunk 1]
```diff
@@ -115,7 +116,11 @@ def __sizeof__(self):
 
     def __reduce_ex__(self, protocol):
         header, frames = device_serialize(self)
-        frames = [f.obj for f in frames]
+        # Since pickle cannot handle memoryviews, we convert them
+        # to NumPy arrays (zero-copy).
+        frames = [
+            (numpy.asarray(f) if isinstance(f, memoryview) else f) for f in frames
+        ]
```
Comment (wence-) [inline dask_cuda/device_host_file.py:123]: What is happening here? AKA how did this ever work if this change is needed?

Comment (madsbk) [reply dask_cuda/device_host_file.py:123]: It assumed that `[f.obj for f in frames]` would convert each frame to a numpy array. This is not the case after https://github.com/rapidsai/cudf/pull/11447

Comment (wence-) [reply dask_cuda/device_host_file.py:123]: Thanks

**Review by wence- — APPROVED** [review]

**Comment by quasiben** [comment]
@gpucibot merge

**Comment by pentschev** [comment]
Thank you @madsbk !

**Comment by jakirkham** [comment]
Thanks Mads! 🙏

**Pull request merged**